lib/cartopy/io/ogc_clients.py

```diff
@@ -511,6 +526,26 @@ def fetch_raster(self, projection, extent, target_resolution):
 
         return located_images
 
+    @property
+    def _cache_dir(self):
+        """Return the name of the cache directory"""
+        return os.path.join(
+            self.cache_path,
+            self.__class__.__name__
+        )
```

Comment:
If you take the pathlib suggestion above this can be written as
```suggestion
        return self.cache_path / self.__class__.__name__
```

Below this you can do cachedir.exists() and similar as well.